088-rebalance-progress-status

```diff
@@ -0,0 +1,328 @@
+
+## Motivation
+
+Information surrounding the progress of an executing partition rebalance is useful for planning future cluster operations. Knowing things like:
+
+- How much time an ongoing partition rebalance has left to take
+- How much data an ongoing partition rebalance has left to transfer 
+
+helps users understand the cost of an ongoing partition rebalance, decide whether or not they should continue or cancel it, and know when future operations will be able to be safely executed.
+
+Further, having this information readily available and easily accessible via `KafkaRebalance` custom resources allows users and third-party tools like the Kubernetes CLI or Strimzi Console to easily track the progression of a partition rebalance. 
+
+## Proposal
+
+This proposal would extend the status section of the `KafkaRebalance` custom resource to include a “progress” section that displays information related to ongoing partition rebalances.
+
+In this “progress” section, we include the following fields:
+
+- estimatedTimeToCompletion: The minimum estimated amount time it will take in minutes until partition rebalance is complete. 
+- percentageComplete: The percentage of the partition rebalance that is completed e.g. values in the range [0-100]%
+- rebalanceProgressConfigMap: The ConfigMap where “non-verbose” JSON payload from Executor State from CruiseControlState endpoint is stored.
+
+### Supported KafkaRebalance States
+
+For initial implementation we will focus on including the “progress” section only in the following KafkaRebalance states:
+
+- “Rebalancing” 
+- "Stopped"
+
+These are the states where this progress information will be able to be most accurately calculated and most useful for users. 
+We could provide the “progress” section for other states as well such as the “ProposalReady” and “Ready” states but it is not completely necessary, nor is it trivial. 
+Further explanation as to why that is and why it should be saved as a future improvement is explained in the Future Improvements section near the bottom of this proposal.
+
+All information required for estimating the values of  “estimatedTimeToCompletion” and  “percentageComplete” fields can be derived from either Cruise Control server configurations or CruiseControlState endpoint. 
+That being said, the method of estimation for these fields depends on the state of the KafkaRebalance resource.
+
+#### estimatedTimeToCompletion
+
+##### Rebalancing
+
+```
+rate = (finishedDataMovement)/(<task_trigger_time> - <current_time>)
+
+estimatedTimeToCompletion = (totalDateToMove-finishedDataMovement) / (rate)
+```
+
+##### Stopped
+
+Once a rebalance has been stopped, it cannot be completed. 
+Therefore, there is no “estimationTimeToCompletion” for a stopped rebalance, so we set estimatedTimeToCompletion = null to emphasize this. 
+The `KafkaRebalance` resource must be refreshed and the progress section overwritten with the next state change. 
```

Comment:
"must be" - Do you mean "will be", as part of the implementation of this proposal or "should be" done be something else?

Reply:
Reworded to make this more clear